### PR TITLE
Added cheatsheet for the idris REPL.

### PR DESCRIPTION
**Pull request by jfdm**
# Cheat sheet for the Idris REPL.

**What does your Instant Answer do?**

A cheatsheet for the Idris prompt. Although the information is available from a command inside the prompt this cheatsheet provides an alternate view.

**What is the data source for your Instant Answer? (Provide a link if possible)**

The data source is the help documentation for the idris real, accessible from the program itself.

More information about Idris is [available online](http://www.idris-lang.org)

**Why did you choose this data source?**

These are the official sources of information.

**Are there any other alternative (better) data sources?**

No.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

For developers, researchers, and enthusiasts of Idris.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**

Not that I am aware of.

**Are you having any problems? Do you need our help with anything?**

No.

**Where did you hear about DuckDuckHack? (For first time contributors)**

From colleagues, you social media links, and the web.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
screen shot 2015-08-01 at 23 30 27

-----
IA Page: https://duck.co/ia/view/idris_repl_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/idris-repl.json [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            {
+                "key" : "[:x] [<expr>]",
+                "val" :  "Execute IO actions resulting from an expression using the interpreter"
+            },
+            {
+                "key" : "[:c] [:compile] [<filename>]",
+                "val" :  "Compile to an executable [codegen] [<filename>]"
+            },
+            {
+                "key" : "[:exec] [:execute] [\\[<expr>]\\]]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/idris-repl.json]: I think you have an extra `]

File: share/goodie/cheat_sheets/idris-repl.json [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            {
+              "key" : "[:mkdoc] [<namespace>]",
+              "val" :  "Generate IdrisDoc for namespace(s) and dependencies"
+            },
+            {
+                "key" : "[:apropos] [<package list>] [<name>]",
+                "val" : "Search names, types, and documentation"
+            },
+            {
+                "key" : "[:s] [:search] [\\[<package list>]\\]] [<expr>]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/idris-repl.json]: Looks like another extra `]`

File: share/goodie/cheat_sheets/idris-repl.json [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            {
+                "key" : "[:unset] [<option>] ",
+                "val" : "Unset an option"
+            },
+            {
+                "key" : "[:color] [:colour] [<option>]",
+                "val" :  "Turn REPL colours on or off; set a specific colour"
+            },
+            {
+                "key" : "[:consolewidth]  [auto|infinite>|number]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/idris-repl.json]: Is the `>` supposed to be there?

Let's format these sub-alternatives properly:

```json
"key" : "[:consolewidth]  ([auto]/[infinite]/[number])",
```

**Comment by jfdm** [comment]
Cheers for the numpty checks. Will update ASAP.

**Comment by moollaza** [comment]
@jfdm I've continued this in #1572

**Comment by moollaza** [comment]
@jfdm congrats! It's live: https://duckduckgo.com/?q=idris+repl+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request closed without merging**